src/main/java/io/gravitee/repository/redis/common/RedisConnectionFactory.java

```diff
@@ -36,29 +46,46 @@
 
     private final String propertyPrefix;
 
+    private final static String SENTINEL_HOST_SEPARATOR = ",";
+
     public RedisConnectionFactory(String propertyPrefix) {
         this.propertyPrefix = propertyPrefix + ".redis.";
     }
 
     @Override
     public org.springframework.data.redis.connection.RedisConnectionFactory getObject() throws Exception {
-        JedisConnectionFactory jedisConnectionFactory = new JedisConnectionFactory();
-        jedisConnectionFactory.setHostName(readPropertyValue(propertyPrefix + "host", String.class, "localhost"));
-        jedisConnectionFactory.setPort(readPropertyValue(propertyPrefix + "port", int.class, 6379));
-        jedisConnectionFactory.setPassword(readPropertyValue(propertyPrefix + "password", String.class, null));
-        jedisConnectionFactory.setTimeout(readPropertyValue(propertyPrefix + "timeout", int.class, -1));
-        jedisConnectionFactory.setUseSsl(readPropertyValue(propertyPrefix + "ssl", boolean.class, false));
-
-        int poolMax = readPropertyValue(propertyPrefix + "pool.max", int.class, 256);
-
-        JedisPoolConfig poolConfig = new JedisPoolConfig();
-        poolConfig.setMaxTotal(poolMax);
-        poolConfig.setBlockWhenExhausted(false);
-        jedisConnectionFactory.setPoolConfig(poolConfig);
-
-        jedisConnectionFactory.afterPropertiesSet();
-
-        return jedisConnectionFactory;
+        final LettuceConnectionFactory lettuceConnectionFactory;
+
+        final String sentinelNodesProperty = readPropertyValue(propertyPrefix + "sentinel.nodes", String.class, "");
```

Comment:
I'm wondering if it couldn't be more relevant to avoid having array in a single String, at least to follow what've been done for the MongoDB repository.

Please have a look to https://github.com/gravitee-io/gravitee-repository-mongodb/blob/master/src/test/resources/graviteeTest.yml#L4

What do you think?

Reply:
I admit that I was inspired by the syntax used in Spring Boot :)  
But no problem to use a yaml list to be consistent with the Mongo configuration.